app/scripts/modules/core/src/pipeline/config/stages/stage.module.js

```diff
@@ -132,6 +135,31 @@ module(CORE_PIPELINE_CONFIG_STAGES_STAGE_MODULE, [
         });
       };
 
+      $scope.getApplicationPermissions = function() {
+        ApplicationReader.getApplicationPermissions($scope.application.name).then(result => {
+          appPermissions = result;
+          if (appPermissions) {
+            const readArray = appPermissions.READ || [];
+            const writeArray = appPermissions.WRITE || [];
+            const executeArray = appPermissions.EXECUTE || [];
+            appRoles = readArray.concat(writeArray, executeArray);
+            appRoles = Array.from(new Set(appRoles));
+            $scope.updateAvailableStageRoles();
```

Comment:
Lodash's `union` method would be helpful here: https://lodash.com/docs/4.17.15#union